deploy/crds/crd-certificaterequests.yaml

```diff
@@ -81,6 +81,19 @@ spec:
                 duration:
                   description: The requested 'duration' (i.e. lifetime) of the Certificate. This option may be ignored/overridden by some issuer types.
                   type: string
+                extra:
+                  description: Extra contains extra attributes of the user that created the CertificateRequest. Populated by the cert-manager webhook on creation and immutable.
```

Comment:
One thing that I noticed whilst experimenting with this is that if I _do_ manually populate these fields on a `CertificateRequest` and apply it, it will succeed first time, since we set them in the mutation function that is run on creation, but fail if I run the apply second time as the mutation function that we run on update don't touch those fields.

I guess this isn't really an issue as usually people don't explicitly apply CRs.

I don't know if it's worth to add a more definite 'Do not try to set these fields..' description.

Reply:
Perhaps, though we can also add this to the docs.